### PR TITLE
feat: add editor and split-pane

### PR DESCRIPTION
**Pull request by PriyanshuDangi**

### CODE REVIEW

File: src/config/config.example.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export const BASE_PATH = 'http://localhost:8080';
+export const BASE_PATH = 'http://localhost:8000/';
```
Comment (CaptainIRS) [inline src/config/config.example.ts:1]: revert

File: package.json [hunk 2]
```diff
@@ -27,27 +27,34 @@
...
-    "react-bootstrap": "^2.1.1",
+    "react-ace": "^9.5.0",
+    "react-bootstrap": "^2.1.2",
     "react-dom": "^17.0.2",
     "react-flags-select": "^2.1.2",
     "react-google-recaptcha": "^2.1.0",
     "react-hot-toast": "^2.2.0",
     "react-paginate": "^8.1.0",
     "react-redux": "^7.2.6",
+    "react-reflex": "^4.0.3",
```
Comment (CaptainIRS) [inline package.json:51]: is this needed?

**Review by CaptainIRS — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (KSIX2) [reply package.json:51]: Sorry, forgot that one. I'll make all the changes

File: src/components/SideBar/SideBar.tsx [hunk 3]
```diff
@@ -15,8 +19,8 @@ import { logout, isloggedIn } from '../../store/User/UserSlice';
 import { useAppDispatch, useAppSelector } from '../../store/hooks';
 import { IconProp } from '@fortawesome/fontawesome-svg-core';
 const icons = [
-  { icon: faCode, route: 'editor', tooltip: 'Code Editor' },
-  { icon: faGlobeAsia, route: 'mapdesigner', tooltip: 'Map Designer' },
+  { icon: faCode, route: 'dashboard', tooltip: 'Code Editor' },
+  { icon: faGlobeAsia, route: 'mapdesginer', tooltip: 'Map Designer' },
```
Comment (CaptainIRS) [inline src/components/SideBar/SideBar.tsx:23]:
```suggestion
  { icon: faGlobeAsia, route: 'mapdesigner', tooltip: 'Map Designer' },
```

**Review by CaptainIRS — APPROVED** [review]

**Pull request merged**